app/helpers/application_helper.rb

```diff
@@ -22,4 +22,11 @@ def render_flash_message(flash_value)
     end
   end
 
+  def user_has_membership_application?
+    current_user.try(:membership_applications).try(:any?) && current_user.try(:membership_applications).try(:last).try(:persisted?)
+  end
+
+  def member_has_company?
+    current_user.is_member && !current_user.membership_applications.last.company.nil?
+  end
```

Comment:
The company needs to be associated with the same company number provided in the application, so imo that field should grab the info from there. 

For the categories they have to be discussed separately..